core/src/Streamly/Internal/Data/Parser/ParserD.hs

```diff
@@ -1700,16 +1778,97 @@ wordQuotedBy keepQuotes isEsc isBegin isEnd toRight isSep
 
+-- | Quote and bracket aware word splitting. Like 'wordBy' but word separators
+-- within specified quotes or brackets are ignored. If the end quote is
+-- different from the start quote it is called a bracket. The following quoting
+-- rules apply:
+--
+-- * A quoting or bracket character can be used within the same quote or
+-- bracket if escaped using the supplied escape char.
+-- * A quoting character other than the current quote can be used inside quotes
+-- without escaping e.g. "a single quote ' inside a double quote".
+-- * A starting quote or bracket char without an ending char generates a parse
+-- error.
+-- * An ending bracket char without a corresponding begin quote is ignored e.g.
+-- this parenthesis ) is ignored.
+-- * Brackets can be nested.
+--
+-- Examples of quotes - "double quotes", 'single quotes', (parens), {braces},
+-- ((nested) brackets).
+--
+-- Example:
+--
+-- >>> :{
+-- >>> q x =
+-- >>>     case x of
+-- >>>         '"' -> Just x
+-- >>>         '\'' -> Just x
+-- >>>         _ -> Nothing
+-- >>> :}
+--
+-- >>> p = Parser.wordWithQuotes (== '\\') q isSpace Fold.toList
+-- >>> Stream.parse p $ Stream.fromList "a\"b'c\";'d\"e'f ghi"
+-- Right "a\"b'c\";'d\"e'f"
+--
+-- Note that outer quotes and backslashes from the input string are consumed by
+-- Haskell, therefore, the actual input string passed to the parser is:
+-- a"b'c";'d"e'f ghi
+--
+-- Similarly, when printing, double quotes are escaped by Haskell.
+--
+{-# INLINE wordWithQuotes #-}
+wordWithQuotes :: (Monad m, Eq a) =>
+       (a -> Bool)     -- ^ Matches an escape elem?
+    -> (a -> Maybe a)  -- ^ If left quote, return right quote, else Nothing.
+    -> (a -> Bool)     -- ^ Matches a word separator?
+    -> Fold m a b
+    -> Parser a m b
+wordWithQuotes = wordWithQuotesWith (const False) True
+
+-- | Like 'wordWithQuotes' but strips the quotes and escape chars.
+--
+-- Example:
+--
+-- >>> :{
+-- >>> q x =
+-- >>>     case x of
+-- >>>         '"' -> Just x
+-- >>>         '\'' -> Just x
+-- >>>         _ -> Nothing
+-- >>> :}
+--
+-- >>> p = Parser.wordStripQuotes (== '\\') q isSpace Fold.toList
+-- >>> Stream.parse p $ Stream.fromList "a\"b'c\";'d\"e'f ghi"
+-- Right "ab'c;d\"ef"
+--
+-- Note that outer quotes and backslashes from the input string are consumed by
+-- Haskell, therefore, the actual input string passed to the parser is:
+-- a"b'c";'d"e'f ghi
+--
+-- From this the outermost quotes are stripped by the parser, double quotes inside
+-- single quotes or vice-versa are kept.
+--
+-- Similarly, when printing, double quotes are escaped by Haskell.
+--
+{-# INLINE wordStripQuotes #-}
+wordStripQuotes :: (Monad m, Eq a) =>
+       (a -> Bool)     -- ^ Matches an escape elem?
+    -> (a -> Maybe a)  -- ^ If left quote, return right quote, else Nothing.
+    -> (a -> Bool)     -- ^ Matches a word separator?
+    -> Fold m a b
+    -> Parser a m b
+wordStripQuotes = wordWithQuotesWith (const False) False
```

Comment:
Is `wordFramedBy` similar to `wordStripQuotes`?
Can `wordFramedBy` be implemented in terms of `wordStripQuotes`?